app/ui/src/main/java/com/fsck/k9/activity/K9Activity.java

```diff
@@ -29,4 +34,24 @@ public boolean dispatchTouchEvent(MotionEvent event) {
     public void setupGestureDetector(OnSwipeGestureListener listener) {
         mBase.setupGestureDetector(listener);
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSIONS_REQUEST_READ_CONTACTS:
+            case PERMISSIONS_REQUEST_WRITE_CONTACTS: {
+                // If request is cancelled, the result arrays are empty.
```

Comment:
it's nicer to express this in code:

```
boolean permissionWasGranted = grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
if (permissionWasGranted) {
```

same amount of lines, but in a formal syntax :)

Reply:
Sure, but in this form it is extendable. I expect at least runtime permissions for storage to be here soon.

Reply:
Oh, you meant the if statement, changed that.